contracts/nft/erc721m/clones/ERC721MagicDropCloneable.sol

```diff
@@ -0,0 +1,401 @@
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(ERC721MagicDropMetadataCloneable)
+        returns (bool)
+    {
+        return interfaceId == type(IERC721MagicDropMetadata).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /*==============================================================
+    =                      ADMIN OPERATIONS                        =
+    ==============================================================*/
+
+    /// @notice Sets up the contract parameters in a single call.
+    /// @dev Only callable by the owner. Configures max supply, wallet limit, URIs, stages, payout recipient, and provenance.
+    /// @param config A struct containing all setup parameters.
+    function setup(SetupConfig calldata config) external onlyOwner {
+        if (config.maxSupply > 0) {
+            _setMaxSupply(config.maxSupply);
+        }
+
+        if (config.walletLimit > 0) {
+            _setWalletLimit(config.walletLimit);
+        }
+
+        if (bytes(config.baseURI).length > 0) {
+            _setBaseURI(config.baseURI);
+        }
+
+        if (bytes(config.contractURI).length > 0) {
+            _setContractURI(config.contractURI);
+        }
+
+        if (config.allowlistStage.startTime != 0 || config.allowlistStage.endTime != 0) {
+            _setAllowlistStage(config.allowlistStage);
+        }
+
+        if (config.publicStage.startTime != 0 || config.publicStage.endTime != 0) {
+            _setPublicStage(config.publicStage);
+        }
+
+        if (config.payoutRecipient != address(0)) {
+            _setPayoutRecipient(config.payoutRecipient);
+        }
+
+        if (config.provenanceHash != bytes32(0)) {
+            _setProvenanceHash(config.provenanceHash);
+        }
+    }
+
+    /// @notice Sets the configuration of the public mint stage.
+    /// @dev Only callable by the owner. Ensures the public stage does not overlap improperly with the allowlist stage.
+    /// @param stage A struct defining the public stage timing and price.
+    function setPublicStage(PublicStage calldata stage) external onlyOwner {
+        _setPublicStage(stage);
+    }
+
+    /// @notice Sets the configuration of the allowlist mint stage.
+    /// @dev Only callable by the owner. Ensures the allowlist stage does not overlap improperly with the public stage.
+    /// @param stage A struct defining the allowlist stage timing, price, and merkle root.
+    function setAllowlistStage(AllowlistStage calldata stage) external onlyOwner {
+        _setAllowlistStage(stage);
+    }
+
+    /// @notice Sets the payout recipient address for primary sale proceeds (after the protocol fee is deducted).
+    /// @dev Only callable by the owner.
+    /// @param newPayoutRecipient The address to receive future withdrawals.
+    function setPayoutRecipient(address newPayoutRecipient) external onlyOwner {
```

Comment:
Can check against zero address here

although since it's an owner function it probably won't happen.

Reply:
shouldn't be necessary because `_splitProceeds` will revert if payout recipient is 0 address